### PR TITLE
feat: insight viz view props

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Towards https://github.com/PostHog/posthog/issues/15680. We want to add Insights to notebooks but first they need to look good

## Changes

- Adds boolean toggles to show / hide the filters and results on the `InsightViz` node
- Adds an `embedded` option that hides the card wrapping the filters (as it is already contained as part of the notebook)
- Refactored the view props logic because I believe it was incorrectly falling back to `showIfFull` when a falsey value was provided e.g. if `query.showHeader` is false then `const disableHeader = query.showHeader ? !query.showHeader : !showIfFull` works off the `showIfFull` value which could be `true`. What we really want is `const disableHeader = !(query.showHeader ?? showIfFull)` which returns `false`.
- Added some keys because console errors

## How did you test this code?

🙈

### CODE REVIEW

File: frontend/src/queries/nodes/InsightViz/InsightViz.tsx [hunk 1]
```diff
@@ -51,13 +51,14 @@ export function InsightViz({ uniqueKey, query, setQuery, context, readOnly }: In
...
 
     const showIfFull = !!query.full
-    const disableHeader = query.showHeader ? !query.showHeader : !showIfFull
-    const disableTable = query.showTable ? !query.showTable : !showIfFull
-    const disableCorrelationTable = query.showCorrelationTable ? !query.showCorrelationTable : !showIfFull
-    const disableLastComputation = query.showLastComputation ? !query.showLastComputation : !showIfFull
-    const disableLastComputationRefresh = query.showLastComputationRefresh
-        ? !query.showLastComputationRefresh
-        : !showIfFull
+    const disableHeader = !(query.showHeader ?? showIfFull)
```
Comment (pauldambra) [inline frontend/src/queries/nodes/InsightViz/InsightViz.tsx:54]: tempting to suggest flipping all of these so they become `showHeader` , `showTable` etc so you don't have to mentally parse "the header won't be shown if show header is not true or if when it is null then......"

but that's scope creep here 😊

File: frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx [hunk 2]
```diff
@@ -243,7 +243,7 @@ export function ActionFilterRow({
     const suffix = typeof customRowSuffix === 'function' ? customRowSuffix({ filter, index, onClose }) : customRowSuffix
 
     const propertyFiltersButton = (
-        <IconWithCount count={filter.properties?.length || 0} showZero={false}>
+        <IconWithCount key="property-filter" count={filter.properties?.length || 0} showZero={false}>
```
Comment (pauldambra) [inline frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:246]: what does the addition of the `key`s do in this file?

**Review by pauldambra — APPROVED** [review]
Clicked around all insights looks OK to me

File: frontend/src/queries/nodes/InsightViz/InsightViz.tsx (hunk 1 above)
Comment (daibhin) [reply frontend/src/queries/nodes/InsightViz/InsightViz.tsx:54]: Yeah, I tend to agree. I'm unsure why it was the way it is but thought it safer to make minimal changes given it's something the Analytics team are working on. Happy to put a PR together

File: frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx (hunk 2 above)
Comment (daibhin) [reply frontend/src/scenes/insights/filters/ActionFilter/ActionFilterRow/ActionFilterRow.tsx:246]: It was to remove React errors clogging up the console while I was trying to debug some other parts. I think it has to do with the fact that the buttons are all rendered in a loop further down

File: frontend/src/queries/nodes/InsightViz/InsightViz.tsx (hunk 1 above)
Comment (pauldambra) [reply frontend/src/queries/nodes/InsightViz/InsightViz.tsx:54]: Well, don't dig too deep... it's a side quest if anything and you might find my dirty fingers at the base of it anyway 🤣
